### PR TITLE
Add jmespath "filter" option to admin kubernetes.List

### PR DESCRIPTION
**Pull request by asalkeld**
This is so we don't have to get huge results from "oc get" and grep for the values we want.

The following works nicely:
```
curl -k "https://localhost:8443/admin/<resourceID>/kubernetesObjects?kind=configmap&namespace=openshift-apiserver&filter=items\[:2\]" | jq "."
```

Note: requires #528

### CODE REVIEW

File: pkg/frontend/kubeactions/kubeactions.go [hunk 1]
```diff
@@ -161,7 +163,16 @@ func (ka *kubeactions) List(ctx context.Context, oc *api.OpenShiftCluster, group
 		return nil, err
 	}
 
-	return ul.MarshalJSON()
+	if jpath == nil {
+		return ul.MarshalJSON()
+	}
+
+	results, err := jpath.Search(ul)
```
Comment (jim-minter) [inline pkg/frontend/kubeactions/kubeactions.go:170]: please add this functionality to get k8s object, list openshift clusters and list cluster resources for completeness

Comment (jim-minter) [inline pkg/frontend/kubeactions/kubeactions.go:170]: this nearly works but not quite.  If I query `items[0]` I see:

```
{
  "Object": {
     ...
  }
}
```

and I should see

```
{
   ...
}
```

The fix, unpleasant as it is, is to do:

```
	b, err := ul.MarshalJSON()
	if err != nil {
		return err
	}

	var o interface{}
	err = json.Unmarshal(b, &o)
	if err != nil {
		return err
	}

	o, err = jpath.Search(o)
	if err != nil {
		return err
	}

	return json.Marshal(o)
```

Comment (asalkeld) [reply pkg/frontend/kubeactions/kubeactions.go]: should be good now..

File: pkg/frontend/frontend.go [hunk 2]
```diff
@@ -289,27 +291,73 @@ func (f *frontend) Run(ctx context.Context, stop <-chan struct{}, done chan<- st
 	}
 }
 
+func adminJmespathFilterValidate(filter string) (*jmespath.JMESPath, error) {
+	if filter == "" {
+		return nil, nil
+	}
+	return jmespath.Compile(filter)
```
Comment (jim-minter) [inline pkg/frontend/frontend.go:298]: suggest returning an *api.CloudError here then no changes needed to adminReply?

Comment (asalkeld) [reply pkg/frontend/frontend.go]: done

**Pull request merged**